async-query/src/main/java/org/opensearch/sql/spark/scheduler/job/ScheduledAsyncQueryJob.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.spark.scheduler.job;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.client.Client;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.jobscheduler.spi.JobExecutionContext;
+import org.opensearch.jobscheduler.spi.ScheduledJobParameter;
+import org.opensearch.jobscheduler.spi.ScheduledJobRunner;
+import org.opensearch.plugins.Plugin;
+import org.opensearch.sql.spark.asyncquery.AsyncQueryExecutorService;
+import org.opensearch.sql.spark.asyncquery.model.NullAsyncQueryRequestContext;
+import org.opensearch.sql.spark.rest.model.CreateAsyncQueryRequest;
+import org.opensearch.sql.spark.rest.model.CreateAsyncQueryResponse;
+import org.opensearch.sql.spark.scheduler.model.ScheduledAsyncQueryJobRequest;
+import org.opensearch.threadpool.ThreadPool;
+
+/**
+ * The job runner class for scheduling async query.
+ *
+ * <p>The job runner should be a singleton class if it uses OpenSearch client or other objects
+ * passed from OpenSearch. Because when registering the job runner to JobScheduler plugin,
+ * OpenSearch has not invoked plugins' createComponents() method. That is saying the plugin is not
+ * completely initialized, and the OpenSearch {@link org.opensearch.client.Client}, {@link
+ * ClusterService} and other objects are not available to plugin and this job runner.
+ *
+ * <p>So we have to move this job runner initialization to {@link Plugin} createComponents() method,
+ * and using singleton job runner to ensure we register a usable job runner instance to JobScheduler
+ * plugin.
+ */
+public class ScheduledAsyncQueryJob implements ScheduledJobRunner {
```

Comment:
The class name sounds weird to me. Should it be something like `ScheduledAsyncQueryJobRunner`?